### PR TITLE
Update arsenal for utf8 tag fix

### PR DESCRIPTION
**Pull request by miniscruff**
# Pull request template

## Description
Downstream fix for utf8 characters in tags from Arsenal

bugfix: S3C-2668

### CODE REVIEW

**Comment by miniscruff** [comment]
@bert-e reset

**Comment by miniscruff** [comment]
@bert-e help

**Comment by miniscruff** [comment]
@bert-e help

**Comment by miniscruff** [comment]
@bert-e help

**Comment by miniscruff** [comment]
@bert-e clear

File: tests/unit/api/objectPut.js [hunk 1]
```diff
@@ -68,11 +60,10 @@ describe('parseTagFromQuery', () => {
         { tagging: 'key1=value1=value2', error: invalidArgument },
         { tagging: '=value1', error: invalidArgument },
         { tagging: 'key1%=value1', error: invalidArgument },
-        { tagging: `${generateString(129)}=value1`, error: invalidTag },
-        { tagging: `key1=${generateString(257)}`, error: invalidTag },
-        { tagging: `${generateString(129)}=value1`, error: invalidTag },
-        { tagging: `key1=${generateString(257)}`, error: invalidTag },
-        { tagging: 'key1#=value1', error: invalidTag },
+        { tagging: `${'w'.repeat(129)}=value1`, error: invalidTag },
```
Comment (ilkescality) [inline tests/unit/api/objectPut.js:63]: good catch

**Review by ilkescality — APPROVED** [review]

**Review by tmacro — APPROVED** [review]

**Review by alexanderchan-scality — APPROVED** [review]

**Review by jonathan-gramain — CHANGES_REQUESTED** [review]
I did a similar comment in one of @ilkescality 's PR: you will need to update the dependency hash on the integration PRs by doing the following on each:

When it's time to resolve conflicts on `w/x.y/...`, update `package.json` with the latest commit hash of the matching dependency branch, eg. the latest `development/8.1` of arsenal when working on `w/8.1/...`, that contains your fix

Then:

```
git reset yarn.lock
git checkout yarn.lock
rm -rf node_modules
yarn install
git add yarn.lock package.json
git commit
```

**Comment by miniscruff** [comment]
> I did a similar comment in one of @ilkescality 's PR: you will need to update the dependency hash on the integration PRs by doing the following on each:
> 
> When it's time to resolve conflicts on `w/x.y/...`, update `package.json` with the latest commit hash of the matching dependency branch, eg. the latest `development/8.1` of arsenal when working on `w/8.1/...`, that contains your fix
> 
> Then:
> 
> ```
> git reset yarn.lock
> git checkout yarn.lock
> rm -rf node_modules
> yarn install
> git add yarn.lock package.json
> git commit
> ```

I went through each of the integration branches and did:
```
git checkout ...
git merge ...
# this failed with a conflict
# resolve conflict to use new hash
yarn install
git add, commit, push
```

This was required because each one had a conflict and failed from 7.6 onwards. I can go through them again but I think this should be done already.

**Comment by jonathan-gramain** [comment]
> > I did a similar comment in one of @ilkescality 's PR: you will need to update the dependency hash on the integration PRs by doing the following on each:
> > When it's time to resolve conflicts on `w/x.y/...`, update `package.json` with the latest commit hash of the matching dependency branch, eg. the latest `development/8.1` of arsenal when working on `w/8.1/...`, that contains your fix
> > Then:
> > ```
> > git reset yarn.lock
> > git checkout yarn.lock
> > rm -rf node_modules
> > yarn install
> > git add yarn.lock package.json
> > git commit
> > ```
> 
> I went through each of the integration branches and did:
> 
> ```
> git checkout ...
> git merge ...
> # this failed with a conflict
> # resolve conflict to use new hash
> yarn install
> git add, commit, push
> ```
> 
> This was required because each one had a conflict and failed from 7.6 onwards. I can go through them again but I think this should be done already.

In `package.json`, the commit hash is the same for this PR and all integration PRs opened by bert-e. They should reflect their respective arsenal branch after the conflict resolution, which is a different hash at least between 7.4 and 8.1 (I think arsenal has less branches than cloudserver, e.g. no development/8.2, in which case bert-e should be able to merge automatically in `w/8.2` from `w/8.1` without manual conflict resolution).

**Comment by miniscruff** [comment]
> > > I did a similar comment in one of @ilkescality 's PR: you will need to update the dependency hash on the integration PRs by doing the following on each:
> > > When it's time to resolve conflicts on `w/x.y/...`, update `package.json` with the latest commit hash of the matching dependency branch, eg. the latest `development/8.1` of arsenal when working on `w/8.1/...`, that contains your fix
> > > Then:
> > > ```
> > > git reset yarn.lock
> > > git checkout yarn.lock
> > > rm -rf node_modules
> > > yarn install
> > > git add yarn.lock package.json
> > > git commit
> > > ```
> > 
> > 
> > I went through each of the integration branches and did:
> > ```
> > git checkout ...
> > git merge ...
> > # this failed with a conflict
> > # resolve conflict to use new hash
> > yarn install
> > git add, commit, push
> > ```
> > 
> > 
> > This was required because each one had a conflict and failed from 7.6 onwards. I can go through them again but I think this should be done already.
> 
> In `package.json`, the commit hash is the same for this PR and all integration PRs opened by bert-e. They should reflect their respective arsenal branch after the conflict resolution, which is a different hash at least between 7.4 and 8.1 (I think arsenal has less branches than cloudserver, e.g. no development/8.2, in which case bert-e should be able to merge automatically in `w/8.2` from `w/8.1` without manual conflict resolution).

oh so use the commit hash from arsenal that matches each development/x version and not the 7.4 version I used each time... got it.

**Review by jonathan-gramain — APPROVED** [review]
Approving in advance since I will be off tomorrow

**Comment by miniscruff** [comment]
Ok for anyone joining late, I went through each of the integration PRs to update the arsenal git hash to the one from the arsenal integration PRs for each version... I think everything is ready now but would not be surprised if I missed something.

**Comment by miniscruff** [comment]
@bert-e reset

**Comment by miniscruff** [comment]
@bert-e force_reset

**Comment by miniscruff** [comment]
@bert-e status

**Comment by miniscruff** [comment]
@bert-e status

**Comment by miniscruff** [comment]
@bert-e status

**Comment by miniscruff** [comment]
ok third try or something... all the integration branches should properly be using the version from arsenal...

**Review by rahulreddy — APPROVED** [review]

**Comment by miniscruff** [comment]
@bert-e status

**Comment by miniscruff** [comment]
@bert-e approve

**Comment by miniscruff** [comment]
@bert-e status

**Comment by miniscruff** [comment]
@bert-e help

File: tests/functional/aws-node-sdk/lib/utility/tagging.js [hunk 2]
```diff
@@ -1,31 +1,15 @@
...
-    return key;
-}
-
 const taggingTests = [
     { tag: { key: '+- =._:/', value: '+- =._:/' },
         it: 'should return tags if tags are valid' },
     { tag: { key: 'key1', value: '' },
         it: 'should return tags if value is an empty string' },
-    { tag: { key: _generateWordWithLength(129), value: 'foo' },
+    { tag: { key: 'w'.repeat(129), value: 'foo' },
```
Comment (rahulreddy) [inline tests/functional/aws-node-sdk/lib/utility/tagging.js:6]: This is a good rewrite 💯

**Pull request merged**